taqueria-state-js/common.ts

```diff
@@ -0,0 +1,71 @@
+import {pipe} from 'rambda'
+import * as Config from "@taqueria/protocol/Config"
+import * as PersistentState from "@taqueria/protocol/PersistentState"
+
+export interface Origination {
+    contract: string
+    address: string
+    destination: string
+    time: number
+}
+
+type PersistedOriginationTaskOutput = Omit<Origination, "time">
+
+export default (projectAbspath: string, config: Config.t, state: PersistentState.t, selectedEnv?: string) => {
+    /**
+     * Gets the configuration for the project
+     * @returns {Config.t}
+     */
+    const getConfig = () => config
+
+
+    /**
+     * Gets the absolute path to the project
+     * @returns {string}
+     */
+    const getProjectAbsPath = () => projectAbspath
+
+    
+    /**
+     * Gets the persistent state of the project
+     * @returns {PersistentState.t}
+     */
+    const getState = () => state
+
+
+    /**
+     * Gets every origination performed for this project, sorted newest to oldest, and
+     * optionally filtered for a particular contract
+     * @returns {Origination[]}
+     */
+    const getOriginations = (contractName?: string) => Object.keys(state.tasks)
```

Comment:
Really like how this works @mweichert! 👌

Reply:
Thanks!